mitodl/open-discussions
# Unpublish runs for edx sources if they are no longer included in API results

…results

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #3920

#### What's this PR do?
- If a course/run is no longer included in the API results used to ingest edx sources (xpro, mitx, mitxonline, open learning library), then unpublish it and its content.
- Does not run `load_content_files` from within `load_run` for edx platforms because this should be done in a separate task

#### How should this be manually tested?
- Set `MIT_ONLINE_*` values to the same as RC in your .env file
- Run `backpopulate_mitxonline_data` followed by `backpopulate_mitxonline_files` if you haven't already
- Run the following query to verify you have published content files:
```python
from course_catalog.models import ContentFile
ContentFile.objects.filter(run__platform="mitxonline", published=True).count()
```

- Create a fake `mitxonline` course:
```
from course_catalog.models import *
fake_course = Course.objects.create(platform="mitxonline", course_id="fake_course", published=True) #fake course
LearningResourceRun.objects.create(platform="mitxonline", run_id="fake_run_1", published=True, object_id=fake_course.id) # Fake run for fake course
LearningResourceRun.objects.create(platform="mitxonline", run_id="fake_run_2", published=True, object_id=<real_course_id>) #fake run for real course
```
- Run `backpopulate_mitxonline_data`.  When it is done, the above course and runs should all be `published=False`
- Run the same content query again, you should have the same number of published content files.
